integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java

```diff
@@ -0,0 +1,618 @@
+            true, "/bin/sh", "-c", "chmod +x " + destLocation),
+        String.format("Failed to change permissions for file %s in pod %s", destLocation, server1Name));
+    assertTrue(execResult.exitValue() == 0,
+        String.format("Failed to change file %s permissions, stderr %s stdout %s", destLocation,
+            execResult.stderr(), execResult.stdout()));
+    logger.info("File permissions changed inside pod");
+
+    /* First, kill the managed server process in the container three times to cause the node manager to
+     * mark the server 'failed not restartable'. This in turn is detected by the liveness probe, which
+     * initiates a container restart.
+     */
+    for (int i = 0; i < 3; i++) {
+      execResult = assertDoesNotThrow(() -> execCommand(domainNamespace, server1Name, null,
+              true, "/bin/sh", "-c", destLocation + " " + server1Name),
+          String.format("Failed to execute script %s in pod %s namespace %s", destLocation,
+              server1Name, domainNamespace));
+      logger.info("Command executed to kill server inside pod, exit value {0}, stdout {1}, stderr {2}",
+          execResult.exitValue(), execResult.stdout(), execResult.stderr());
+
+      try {
+        Thread.sleep(2 * 1000);
+      } catch (InterruptedException ie) {
+        // ignore
+      }
+    }
+
+    // check pod is ready
+    checkPodReady(server1Name, domainUid, domainNamespace);
+
+    // get the restart count of the container in pod after liveness probe restarts
+    int afterRestartCount = assertDoesNotThrow(() ->
+            getContainerRestartCount(domainNamespace, null, server1Name, null),
+        String.format("Failed to get the restart count of the container from pod {0} in namespace {1}",
+            server1Name, domainNamespace));
+    assertTrue(afterRestartCount - beforeRestartCount == 1,
+        String.format("Liveness probe did not start the container in pod %s in namespace %s",
+            server1Name, domainNamespace));
+
+    // check the sample app is accessible from admin server pod
+    for (int i = 1; i <= replicaCount; i++) {
+      testUntil(
+          checkSampleAppReady(domainUid, domainNamespace, serverNamePrefix + i),
+          logger,
+          "sample app is accessible from server {0} in namespace {1}",
+          serverNamePrefix + i,
+          domainNamespace);
+    }
+
+    // shutdown domain and verify the domain is shutdown
+    shutdownDomainAndVerify(domainNamespace, domainUid, replicaCount);
+  }
+
+  /**
+   * Test rolling restart for a multi-clusters domain.
+   * Make sure pods are restarted only once.
+   * Verify all pods are terminated and restarted only once
+   * Rolling restart triggered by changing:
+   * imagePullPolicy: IfNotPresent --> imagePullPolicy: Never
+   * Verify domain changed event is logged.
+   */
+  @Test
+  @DisplayName("Verify server pods are restarted only once by changing the imagePullPolicy in multi-cluster domain")
```

Comment:
Move the tests into a java class where we verify the  server pod restart on domain attribute change. e.g.  ItPodsRestart

Reply:
There is a method testServerPodsRestartByChangingImagePullPolicy() in ItPodsRestart which tests the same usecase. I will delete this method.